src/WebJobs.Extensions.DurableTask/DurableActivityContext.cs

```diff
@@ -102,6 +103,31 @@ internal object GetInput(Type destinationType)
             }
 
             string serializedValue = jToken.ToString(Formatting.None);
+
+            // Object inputs for out-of-proc activities are passed in their JSON-stringified form
+            // with a destination type of System.String. Unfortunately, deserializing a JSON string
+            // to a string causes MessagePayloadDataConverter to throw an exception. This is a
+            // workaround for that case.
+            if (destinationType.Equals(typeof(string)))
+            {
+                // Json.NET has no TryParse or IsValidJson method, so we'll optimistically parse
+                // and catch any exceptions (which shouldn't occur: all (in-proc; JSON and not JSON)
+                // or all other (out-of-proc: not-JSON) inputs with destination System.String should
+                // cast to valid JValues and be handled above.)
+                try
+                {
+                    var deserializedValue = MessagePayloadDataConverter.Default.Deserialize(serializedValue, typeof(object));
+                    if (deserializedValue != null)

```

Comment:
What does it mean if `deserializedValue` is null?

Reply:
Thanks for catching this; that's a doc read error on my end. JsonSerializer will throw an exception rather than return null if it can't parse a string, but I read it the other way around. Removing as it makes no sense with that correction.